### PR TITLE
feat: add `cosign` autocompletion

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

feature

**What is the current behavior? (You can also link to an open issue here)**

no support for cosign

**What is the new behavior (if this is a feature change)?**

will add cosign CLI autocompletion support

**Additional info:**

https://github.com/sigstore/cosign
https://github.com/sigstore/cosign/pull/1360

### CODE REVIEW

**Review by mschrage — COMMENTED** [review]
I haven't looked too closely at this but overall it looks good. One thing to be aware of is that since `cosign` is implemented using `cobra`, it should be possible to autogenerate a completion spec. 

Check out [this guide](https://fig.io/docs/guides/autocomplete-for-teams/cobra) for more details.

**Comment by developer-guy** [comment]
> I haven't looked too closely at this but overall it looks good. One thing to be aware of is that since `cosign` is implemented using `cobra`, it should be possible to autogenerate a completion spec.
> 
> Check out [this guide](https://fig.io/docs/guides/autocomplete-for-teams/cobra) for more details.

Yep, I've already opened a PR to cosign side 👇
https://github.com/sigstore/cosign/pull/1360

**Comment by developer-guy** [comment]
kindly ping 🙋🏻‍♂️

**Comment by fedeci** [comment]
I imagine this has been generated by adding the cobra integration locally, right?

File: src/cosign.ts [hunk 1]
```diff
@@ -0,0 +1,3779 @@
...
+      subcommands: [
+        {
+          name: "attestation",
+          description: "Attach attestation to the supplied container image",
+          options: [
+            {
+              name: "--output-file",
+              description: "Log output to a file",
+              displayName: "output-file",
+              args: { name: "output-file" },
```
Comment (fedeci) [inline src/cosign.ts]: This accepts a filepaths template.

File: src/cosign.ts [hunk 2]
```diff
@@ -0,0 +1,3779 @@
...
+            },
+            {
+              name: "--sbom",
+              description: "Path to the sbom, or {-} for stdin",
+              displayName: "sbom",
+              args: { name: "sbom" },
+            },
+            {
+              name: "--type",
+              description: "Type of sbom (spdx|cyclonedx|syft)",
```
Comment (fedeci) [inline src/cosign.ts]: These suggestions can be mentioned in the `suggestions` prop of arg.

**Review by fedeci — COMMENTED** [review]
I am pretty sure this has been generated via the cobra integration. There are a lot of lines here and it would be quite long to review all the possible templates/suggestions for the args both for us and for you.
What we could do to add useful props quickly is to perform a strategic search of some keywords in the spec.
e.g.:
- options that has output,input,path in the name will likely accept filepaths
- `/\([a-zA-Z0-9]+\|/g` this regex will instead highlight all options that require a suggestions field)

**Review by fedeci — APPROVED** [review]
Thanks!

**Comment by developer-guy** [comment]
thanks for merging @fedeci

**Pull request merged**